vllm/compilation/decorators.py

```diff
@@ -0,0 +1,84 @@
+from typing import List, Optional, Union
+
+import torch
+
+import vllm.envs as envs
+from vllm.attention import AttentionMetadata
+from vllm.compilation.levels import CompilationLevel
+from vllm.compilation.wrapper import TorchCompileWrapperWithCustomDispatcher
+from vllm.sequence import IntermediateTensors
+
+
+def support_compile_llama_style(cls: type):
```

Comment:
I think it would be good to have a more general decorator, similar to pytorch `custom_op`, e.g.

```
@support_compilation(dynamic_arg_dims=[("input_ids", [0]), ("positions", [0]), ("inputs_embeds, [0]), ("intermediate_tensors", [0])])
```

I think this would make it more obvious what was going on for anyone looking at the code,  especially if an argument name needs to be changed or arguments to the decorated function are being added/removed/shuffled.

Reply:
it cannot be general, because we need the `__call__` signature to exactly match the function.

if you define a general `def __call__(self, *args, **kwargs)` , how can you parse `input_ids` if people use position arguments to call the function?

Reply:
`custom_op` must deal with something similar.  Maybe the `inspect.signature` can be used on `cls.forward`?   `torch._library.infer_schema` seems relevant.

Reply:
I don't think it would be too unreasonable to make some assumptions about common types (`List`, `Optional`, `Dict`) or builtin vllm types (`IntermediateTensors`) but I guess we can revisit this later.